### DISCUSSION TITLE
車体を旋回させたときにzx120のクローラも一緒に動く問題

### DISCUSSION THREAD

**Discussion by Yamayou-Omiya** [pwri-opera/OperaSim-PhysX#56] [Q&A]
質問失礼します．
zx120のモデルを以下のように動かした場合，クローラも一緒に動作してしまうのですが，どうすれば旋回時にクローラが動かないようにできますか．

https://github.com/user-attachments/assets/1c0b442c-6ace-41c0-9a6e-4a847bd93b59

**Comment by yosuke** [comment]
@Yamayou-Omiya 
質問に気づかず、返信がかなり遅れてしまいすみません。

クローラーにはUnityのWheelColliderを使っているのですが、WheelColliderと建機本体のArticulationBodyを接続するためにFixedJointを設定しています。UnityのFixedJointは「固定関節」を意図した機能であるものの、リンクが重くてトルクが大きい場合は、トルクに負けて動いてしまうことがあるらしく、動画で報告いただいたような挙動が出てしまいます。

下のURLの記事にあるように、[Edit]->[Project Settings]->[Physics]->[Default Solver Iterations] で変更できるパラメータを最大の「255」に変更すると、本症状についてはある程度は軽減できます。
http://blog.aohatablau.net/?eid=2

ただ、大きなトルクをかけると上記のパラメータ調整をしても多少振られてしまうため、より根本的な解決法を現在調査しています。

解決法がわかったら、またすぐにご連絡します。

**Reply by Yamayou-Omiya** [reply]
返信ありがとうございました．こちらでも色々試してみたいと思います．
